app/components/language-select.js

```diff
@@ -0,0 +1,39 @@
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+  config: Ember.inject.service(),
+  i18n: Ember.inject.service(),
+  languageOptions: Ember.computed(function() {
+    return Object.keys(this.get('languageMap'));
+  }),
+  onFinish: null,
+  languageMap: Ember.computed(() => {
```

Comment:
Fixed.  Duplicated code removed and language names localized.  One callout- I was able to remove the `value` attribute from the select-list without any apparent adverse effects.  Not sure why this was the case, since I thought `value` maps to the `value` attribute of the `option` tag.  Does this get overridden if the `optionValuePath` and `optionLabelPath` attrs are used instead?